resources/user.py

```diff
@@ -154,11 +154,15 @@ def post(self):
 class UsersRole(Resource):
     parser = reqparse.RequestParser()
     parser.add_argument('userrole',type=int,required=True,help="This field cannot be blank.")
+    parser.add_argument('name',type=str,required=False)
 
     @admin_required
     def post(self):
         data = UsersRole.parser.parse_args()
-        users = UserModel.find_by_role(RoleEnum(data['userrole']))
+        if data["name"]:
+            users = UserModel.find_by_role_and_name(RoleEnum(data['userrole']), data['name'])
+        else:
+            users = UserModel.find_by_role(RoleEnum(data['userrole']))
```

Comment:
I'm curious if the else branch of this will ever evaluate. Depending on how the front end uses/implements this endpoint it might never evaluate. @Chris-Boe what do you think?

Reply:
The else-branch was pre-existing, the UI calls it for a couple different userrole arguments, so I didn't want to risk breaking that. The if-branch is for a new feature in the Tenants view where you can search users by name in addition to role.

Reply:
This reminds me to ask: would it be good to refactor this to be a GET with query string arguments instead of a POST? Conventionally, POST is for creating a restful resource, while GET is for retrieving.

Reply:
@NickSchimek  Yeah we use this on the dashboard, add-tenant, property managers, and JOIN staff pages to get users by role. I was surprised to see it as a POST when I started adding a call to it, so if you wanted to switch it to a GET that'd be fine with me!

Reply:
Yes, I was suprised too. a GET refactor would be welcomed. 

My concern is that if this is used as a form, and if the user does not fill in the name and it is sent as an empty string then in python an empty string is truthy. Hence the if conditional will always evaluate to true. - However if the frontend sends an empty parameter as None then there wont be an issue.

Reply:
This should be a GET request.